packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -224,14 +225,48 @@ export class FieldSchema<
 	protected _typeCheck?: MakeNominal;
 
 	/**
-	 * @param kind - The [kind](https://en.wikipedia.org/wiki/Kind_(type_theory)) of this field.
-	 * Determine the multiplicity, viewing and editing APIs as well as the merge resolution policy.
-	 * @param allowedTypes - What types of tree nodes are allowed in this field.
+	 * Lazily evaluated allowed types.
 	 */
+	private readonly lazyTypes: Lazy<AllowedTypes>;
+
+	/**
+	 * What types of tree nodes are allowed in this field.
+	 *
+	 * @remarks
+	 * This is the same set of types in {@link FlexFieldSchema.allowedTypes}, just as a set with
+	 * laziness removed.
+	 */
+	public get allowedTypeSet(): AllowedTypes {
+		return this.lazyTypes.value;
+	}
+
 	public constructor(
+		/**
+		 * The {@link https://en.wikipedia.org/wiki/Kind_(type_theory) | kind } of this field.
+		 * Determines the multiplicity, viewing and editing APIs as well as the merge resolution policy.
+		 */
 		public readonly kind: Kind,
+		/**
+		 * What types of tree nodes are allowed in this field.
+		 */
 		public readonly allowedTypes: Types,
-	) {}
+	) {
+		this.lazyTypes = new Lazy(() => FieldSchema.normalizeAllowedTypes(this.allowedTypes));
+	}
+
+	/**
+	 * Normalizes a {@link ImplicitAllowedTypes} to a {@link AllowedTypes}.
+	 */
+	public static normalizeAllowedTypes(types: ImplicitAllowedTypes): AllowedTypes {
+		return isReadonlyArray(types) ? types : [types];
+	}
+
+	/**
+	 * Normalizes a {@link ImplicitFieldSchema} to a {@link FieldSchema}.
+	 */
+	public static normalize(schema: ImplicitFieldSchema): FieldSchema {
```

Comment:
This one I think makes sense being here, but putting it here forces it to be package exported as public, and thus its expanding the stable public API. You may want to include that in the API design review for this work (or move it to be a free function)

Reply:
Yeah, free function is probably the way to go then. At least for now. I prefer not adding something to the API until we discover a real customer need for it.